src/lambda/snsToSlack.ts

```diff
@@ -0,0 +1,83 @@
+import { SNSEvent } from 'aws-lambda';
+import fetch from 'node-fetch';
+
+interface PostDataAttachment {
+  color?: string;
+  text?: string;
+  fields?: {
+    title: string;
+    value: string;
+    short: boolean;
+  }[];
+}
+
+interface SlackHookPostData {
+  username: string;
+  text: string;
+  icon_emoji: string;
+  attachments: PostDataAttachment[];
+}
+
+enum Severity {
+  Good = 'good',
+  Warning = 'warning',
+}
+
+// absolute url
+const slackWebhook = process.env.SLACK_WEBHOOK as string;
+
+export const snsToSlackHandler = async (event: SNSEvent): Promise<void> => {
+  const message = JSON.parse(event?.Records[0]?.Sns?.Message || '{}');
+  const region = event?.Records[0]?.EventSubscriptionArn?.split(':')[3];
+  const { AlarmName } = message;
+  const severity = message?.NewStateValue === 'OK' ? Severity.Good : Severity.Warning;
+  const cwUrl =
+    region &&
+    'https://console.aws.amazon.com/cloudwatch/home?region=' +
+      region +
+      '#alarm:alarmFilter=ANY;name=' +
+      encodeURIComponent(AlarmName);
+
+  const slackMessageFields = {
+    CloudWatchUrl: cwUrl,
+    ...message,
+  };
+
+  const allowedFields = [
+    'CloudWatchUrl',
+    'AlarmDescription',
+    'NewStateReason',
+    'NewStateValue',
+    'OldStateValue',
+    'StateChangeTime',
+    'Region',
+  ];
+
+  const postData: SlackHookPostData = {
+    username: 'AWS CloudWatch Warning',
+    text: '*' + event.Records[0].Sns.Subject + '*',
```

Comment:
Move this to top as variable where other event data is extracted as well
```
const subject = event.Records[0].Sns.Subject;
```
Also use the template literal:
```
`*${subject}*`
```